src/connector/src/sink/snowflake.rs

```diff
@@ -0,0 +1,279 @@
+    #[serde(rename = "snowflake.schema")]
+    pub schema: String,
+
+    /// The created pipe object, will be used as `insertFiles` target
+    #[serde(rename = "snowflake.pipe")]
+    pub pipe: String,
+
+    /// The unique, snowflake provided `account_identifier`
+    /// NOTE: please use the form `<orgname>-<account_name>`
+    /// For detailed guidance, reference: <https://docs.snowflake.com/en/user-guide/admin-account-identifier>
+    #[serde(rename = "snowflake.account_identifier")]
+    pub account_identifier: String,
+
+    /// The user that owns the table to be sinked
+    /// NOTE: the user should've been granted corresponding *role*
+    /// reference: <https://docs.snowflake.com/en/sql-reference/sql/grant-role>
+    #[serde(rename = "snowflake.user")]
+    pub user: String,
+
+    /// The public key fingerprint used when generating custom `jwt_token`
+    /// reference: <https://docs.snowflake.com/en/developer-guide/sql-api/authenticating>
+    #[serde(rename = "snowflake.rsa_public_key_fp")]
+    pub rsa_public_key_fp: String,
+
+    /// The rsa pem key *without* encryption
+    #[serde(rename = "snowflake.private_key")]
+    pub private_key: String,
+
+    /// The s3 bucket where intermediate sink files will be stored
+    #[serde(rename = "snowflake.s3_bucket")]
+    pub s3_bucket: String,
+
+    /// s3 credentials
+    #[serde(rename = "snowflake.aws_access_key_id")]
+    pub aws_access_key_id: String,
+
+    /// s3 credentials
+    #[serde(rename = "snowflake.aws_secret_access_key")]
+    pub aws_secret_access_key: String,
+
+    /// The s3 region, e.g., us-east-2
+    #[serde(rename = "snowflake.aws_region")]
+    pub aws_region: String,
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, WithOptions)]
+pub struct SnowflakeConfig {
+    #[serde(flatten)]
+    pub common: SnowflakeCommon,
+}
+
+impl SnowflakeConfig {
+    pub fn from_hashmap(properties: HashMap<String, String>) -> Result<Self> {
+        let config =
+            serde_json::from_value::<SnowflakeConfig>(serde_json::to_value(properties).unwrap())
+                .map_err(|e| SinkError::Config(anyhow!(e)))?;
+        Ok(config)
+    }
+}
+
+#[derive(Debug)]
+pub struct SnowflakeSink {
+    pub config: SnowflakeConfig,
+    schema: Schema,
+    pk_indices: Vec<usize>,
+    is_append_only: bool,
+}
+
+impl Sink for SnowflakeSink {
+    type Coordinator = DummySinkCommitCoordinator;
+    type LogSinker = LogSinkerOf<SnowflakeSinkWriter>;
+
+    const SINK_NAME: &'static str = SNOWFLAKE_SINK;
+
+    async fn new_log_sinker(&self, writer_param: SinkWriterParam) -> Result<Self::LogSinker> {
+        Ok(SnowflakeSinkWriter::new(
+            self.config.clone(),
+            self.schema.clone(),
+            self.pk_indices.clone(),
+            self.is_append_only,
+        )
+        .await
+        .into_log_sinker(writer_param.sink_metrics))
+    }
+
+    async fn validate(&self) -> Result<()> {
+        Ok(())
```

Comment:
It looks like only append-only is supported, please check it here. Schema, conn, and other issues at create time, we can also check it.

Reply:
`schema` may not be possible to check at present, but let me do some further investigations.